src/test/coins_tests.cpp

```diff
@@ -55,10 +55,10 @@ class CCoinsViewTest : public CCoinsView
 
     uint256 GetBestBlock() const override { return hashBestBlock_; }
 
-    bool BatchWrite(CCoinsMap& mapCoins, const uint256& hashBlock, bool erase = true) override
+    bool BatchWrite(CoinsCachePair* pairs, const uint256& hashBlock, bool will_erase = true) override
     {
-        for (CCoinsMap::iterator it = mapCoins.begin(); it != mapCoins.end(); it = erase ? mapCoins.erase(it) : std::next(it)) {
-            if (it->second.flags & CCoinsCacheEntry::DIRTY) {
+        for (auto it{pairs}; it != nullptr; it = it->second.Next(will_erase || !it->second.coin.IsSpent())) {
+            if (it->second.GetFlags() & CCoinsCacheEntry::DIRTY) {
```

Comment:
this `it->second.GetFlags() & CCoinsCacheEntry::DIRTY` appears quite often, maybe it would make sense to extract it to `it->second.IsDirty()` (and `it->second.IsFresh()`, of course)

Reply:
I think this makes the diff easier to follow, since in git it highlights only the `flags` and `GetFlags()` in the line.

Reply:
if there was a separate commit doing only this, it would simplify the other important commits a lot

Reply:
Done.